Neos.Flow/Classes/Http/Client/CurlEngine.php

```diff
@@ -137,7 +137,13 @@ public function sendRequest(Http\Request $request)
             curl_setopt($curlHandle, CURLOPT_PORT, $requestUri->getPort());
         }
 
-        // CURLOPT_COOKIE
+        if (count($request->getCookies()) > 0) {
+            $cookies = array();
+            foreach ($request->getCookies() as $cookie) {
+                $cookies[] = $cookie->getName() . '=' . $cookie->getValue();
+            }
+            curl_setopt($curlHandle, CURLOPT_COOKIE, implode(';', $cookies));
```

Comment:
I think the glue needs to be `'; '` (see https://forums.phpfreaks.com/topic/77789-solved-multiple-cookies-in-curl-curlopt-cookies/) and unsure if a delimiting semicolon is required, but the CURL documentation (https://curl.haxx.se/libcurl/c/CURLOPT_COOKIE.html) has it.

Reply:
you are right on the glue, there has to be a space.
according to the RFC there is no need for a delimiting semicolon.
https://tools.ietf.org/html/rfc6265#section-4.2.1
I hope I will find some time in the near future to provide a test for this feature.